proxy/template.go

```diff
@@ -54,6 +54,16 @@ func getFrontTemplate(s Service) string {
         {{- end}}
     {{- end}}
 {{- end}}
+{{- if $.RedirectWhenNotHttpsProto}}
+    {{- range .ServiceDest}}
+        {{- if ne .ReqMode "https"}}
```

Comment:
`ReqMode` is never `https`. It's always `http` (or `tcp`), so that line will always be true.